### PR TITLE
feat: added subscription group in braze

### PR DESCRIPTION
**Pull request by aashishmalik**
## Description of the change

> Subscription Group support in braze

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/braze/transform.js [hunk 1]
```diff
@@ -349,6 +349,64 @@ function processGroup(message, destination) {
...
+    );
+    response.body.JSON = removeUndefinedValues(subscriptionGroup);
+    return {
+      ...response,
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+        Authorization: `Bearer ${destination.Config.restApiKey}`
+      }
+    };
```
Comment (ItsSudip) [inline v0/destinations/braze/transform.js:407]: We should use `buildResponse` function here.

File: v0/destinations/braze/transform.js [hunk 2]
```diff
@@ -349,6 +349,64 @@ function processGroup(message, destination) {
...
+        })
+        .build();
+    }
+    subscriptionGroup.subscription_state = message.traits.subscriptionState;
+    subscriptionGroup.external_id = [message.userId || message.anonymousId];
+    if (getFieldValueFromMessage(message, "phone")) {
+      subscriptionGroup.phone = [getFieldValueFromMessage(message, "phone")];
+    } else if (getFieldValueFromMessage(message, "email")) {
+      subscriptionGroup.email = [getFieldValueFromMessage(message, "email")];
+    }
```
Comment (ItsSudip) [inline v0/destinations/braze/transform.js:394]:
```suggestion
    const phone = getFieldValueFromMessage(message, "phone");
    const email = getFieldValueFromMessage(message, "email");
    if (phone) {
      subscriptionGroup.phone = phone;
    } else if (email) {
      subscriptionGroup.email = email;
    }
```

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/destinations/braze/transform.js (hunk 2 above)
Comment (aashishmalik) [reply v0/destinations/braze/transform.js]: done

[the remark this answers is not in the document]
Comment (aashishmalik) [reply v0/destinations/braze/transform.js:398]: one of them is mandatory, throwing error if not passed in traits

File: v0/destinations/braze/transform.js [hunk 3]
```diff
@@ -349,6 +349,64 @@ function processGroup(message, destination) {
...
+        .setStatTags({
+          destType: DESTINATION,
+          scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+          stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+          meta:
+            TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.BAD_PARAM
+        })
+        .build();
+    }
+    subscriptionGroup.subscription_state = message.traits.subscriptionState;
```
[the remark this answers is not in the document]
Comment (aashishmalik) [reply v0/destinations/braze/transform.js:390]: I think it is not required as I am checking for message.traits in begining

File: v0/destinations/braze/transform.js [hunk 4]
```diff
@@ -349,6 +349,64 @@ function processGroup(message, destination) {
...
+    const subscriptionGroup = {};
+    subscriptionGroup.subscription_group_id = groupId;
+    if (
+      message.traits.subscriptionState !== "subscribed" &&
+      message.traits.subscriptionState !== "unsubscribed"
+    ) {
+      throw new ErrorBuilder()
+        .setStatus(400)
+        .setMessage(
+          "You must provide a subscription state that is either subscribed or unsubscribed. Aborting message."
```
[the remark this answers is not in the document]
Comment (aashishmalik) [reply v0/destinations/braze/transform.js]: Done

File: v0/destinations/braze/transform.js (hunk 1 above)
Comment (aashishmalik) [reply v0/destinations/braze/transform.js:411]: for this use case case, this is good I believe as we are fetching diff endpoint and format is diff in buildReponse method

**Review by ItsSudip — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Review by sandeepdsvs — APPROVED** [review]

**Pull request merged**